src/wire/ipv6routing.rs

```diff
@@ -0,0 +1,664 @@
+    // |                                                               |
+    // .                                                               .
+    // .                        Addresses[1..n]                        .
+    // .                                                               .
+    // |                                                               |
+    // +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+
+    // 8-bit field containing the CmprI and CmprE values.
+    pub const CMPR:  usize = 4;
+    // 8-bit field containing the Pad value.
+    pub const PAD:   usize = 5;
+    // Variable length field containing addresses
+    pub fn ADDRESSES(length_field: u8) -> Field {
+        let data = DATA(length_field);
+        8..data.end
+    }
+}
+
+/// Core getter methods relevant to any routing type.
+impl<T: AsRef<[u8]>> Header<T> {
+    /// Create a raw octet buffer with an IPv6 Routing Header structure.
+    pub fn new(buffer: T) -> Header<T> {
+        Header { buffer }
+    }
+
+    /// Shorthand for a combination of [new] and [check_len].
+    ///
+    /// [new]: #method.new
+    /// [check_len]: #method.check_len
+    pub fn new_checked(buffer: T) -> Result<Header<T>> {
+        let header = Self::new(buffer);
+        header.check_len()?;
+        Ok(header)
+    }
+
+    /// Ensure that no accessor method will panic if called.
+    /// Returns `Err(Error::Truncated)` if the buffer is too short.
+    ///
+    /// The result of this check is invalidated by calling [set_header_len].
+    ///
+    /// [set_header_len]: #method.set_header_len
+    pub fn check_len(&self) -> Result<()> {
+        let len = self.buffer.as_ref().len();
+        if len < field::MIN_HEADER_SIZE {
+            return Err(Error::Truncated);
+        }
+
+        if len < field::DATA(self.header_len()).end as usize {
+            return Err(Error::Truncated);
+        }
+
+        Ok(())
+    }
+
+    /// Consume the header, returning the underlying buffer.
+    pub fn into_inner(self) -> T {
+        self.buffer
+    }
+
+    /// Return the next header field.
+    #[inline]
+    pub fn next_header(&self) -> Protocol {
+        let data = self.buffer.as_ref();
+        Protocol::from(data[field::NXT_HDR])
+    }
+
+    /// Return the header length field. Length of the Routing header in 8-octet units,
+    /// not including the first 8 octets.
+    #[inline]
+    pub fn header_len(&self) -> u8 {
+        let data = self.buffer.as_ref();
+        data[field::LENGTH]
+    }
+
+    /// Return the routing type field.
+    #[inline]
+    pub fn routing_type(&self) -> Type {
+        let data = self.buffer.as_ref();
+        Type::from(data[field::TYPE])
+    }
+
+    /// Return the segments left field.
+    #[inline]
+    pub fn segments_left(&self) -> u8 {
+        let data = self.buffer.as_ref();
+        data[field::SEG_LEFT]
+    }
+
+    /// Return the IPv6 Home Address
```

Comment:
Could type specific packet methods be in separate documented `impl` blocks. See `ndisc.rs`.

Reply:
I had this but thought we were moving away from it with only panic comments. Done.